### ISSUE TITLE
Add ability to turn off subscription key in API Management API 

### ISSUE THREAD

**Issue by brunhil** [terraform-providers/terraform-provider-azurerm#4884]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The **azurerm_api_management_api** resource is missing the ability to set the ability to require a subscription key to false. 

(https://github.com/Azure/azure-sdk-for-go/blob/master/services/apimanagement/mgmt/2018-01-01/apimanagement/models.go#L949).

### New or Affected Resource(s)

* azurerm_api_management_api

### Potential Terraform Configuration

```hcl

resource "azurerm_api_management_api" "test" {
  name                = "acctestapi-%d"
  resource_group_name = "${azurerm_resource_group.test.name}"
  api_management_name = "${azurerm_api_management.test.name}"
  display_name        = "Butter Parser"
  path                = "butter-parser"
  protocols           = ["https", "http"]
  revision            = "3"
  description         = "What is my purpose? You parse butter."
  service_url         = "https://example.com/foo/bar"
  
  authentication_settings {
	 subscription_key_required = false
  }
}
```

**Comment by BernhardRode** [comment]
Any news, when this feature will be release? =)

**Comment by brunhil** [comment]
@BernhardRode, as mentioned in [PR 4885](https://github.com/terraform-providers/terraform-provider-azurerm/pull/4885), this is blocked by [Azure/azure-rest-api-specs#6372](https://github.com/Azure/azure-rest-api-specs/issues/6372). When that gets completed I can pick this up again :)

**Comment by rvilike** [comment]
@brunhil [Azure/azure-rest-api-specs#6372](https://github.com/Azure/azure-rest-api-specs/issues/6372) is done now, as I can see  😊

**Comment by brunhil** [comment]
This was recently released/closed by #4885 .
